### ISSUE TITLE
KeyError: 'instances'  when running nndet_prep on nndet_example dataset

### ISSUE THREAD

**Issue by ndebs** [MIC-DKFZ/nnDetection#26]
Hello everyone,

When using nndet_prep 000D3 command (on the test dataset), the following error occured:

```
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/anaconda/envs/nnDetection/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/anaconda/envs/nnDetection/lib/python3.9/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "/home/ndebs/code/nndetection/nnDetection/nndet/planning/properties/instance.py", line 155, in analyze_instances_per_case
    props["num_instances"] = count_instances(props, all_classes)
  File "/home/ndebs/code/nndetection/nnDetection/nndet/planning/properties/instance.py", line 177, in count_instances
    instance_classes = list(map(int, props["instances"].values()))
KeyError: 'instances'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/anaconda/envs/nnDetection/bin/nndet_prep", line 33, in <module>
    sys.exit(load_entry_point('nndet', 'console_scripts', 'nndet_prep')())
  File "/home/ndebs/code/nndetection/nnDetection/nndet/utils/check.py", line 58, in wrapper
    return func(*args, **kwargs)
  File "/home/ndebs/code/nndetection/nnDetection/scripts/preprocess.py", line 475, in main
    run(OmegaConf.to_container(cfg, resolve=True),
  File "/home/ndebs/code/nndetection/nnDetection/scripts/preprocess.py", line 394, in run
    run_dataset_analysis(cropped_output_dir=Path(cfg["host"]["cropped_output_dir"]),
  File "/home/ndebs/code/nndetection/nnDetection/scripts/preprocess.py", line 198, in run_dataset_analysis
    _ = analyzer.analyze_dataset(properties)
  File "/home/ndebs/code/nndetection/nnDetection/nndet/planning/analyzer.py", line 80, in analyze_dataset
    props.update(property_fn(self))
  File "/home/ndebs/code/nndetection/nnDetection/nndet/planning/properties/instance.py", line 46, in analyze_instances
    props_per_case = run_analyze_instances(analyzer, all_classes)
  File "/home/ndebs/code/nndetection/nnDetection/nndet/planning/properties/instance.py", line 77, in run_analyze_instances
    props = p.starmap(analyze_instances_per_case, zip(
  File "/anaconda/envs/nnDetection/lib/python3.9/multiprocessing/pool.py", line 372, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
  File "/anaconda/envs/nnDetection/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
KeyError: 'instances'
```

I am facing the exact same problem when running nndet_prep on my own personal dataset.  It seems that the key "instances" is not created in the pickle files in "data/Task000D3_Example/raw_cropped/labelsTr/" (I printed the dictionnary inside caseXX.pkl in order to check).

Do you have any idea how to solve this problem?

Best 

Noëlie

**Comment by mibaumgartner** [comment]
Hi @ndebs,

I just reran the example and did not encounter any issues.
The commands I ran:
```
nndet_example
nndet_prep 000
```

The error you describe typically arrises when the json files of the labels in `raw_splitted/labelsTr` don't contain the "instances" key. Running `nndet_prep` with the `--full_check` will check the underlying data and raise an error if something is missing. 

The example script generates the json files here:
https://github.com/MIC-DKFZ/nnDetection/blob/880d40a135e388c230621c544ccdaff1aa99151a/scripts/generate_example.py#L91-L98

Best,
Michael

**Comment by ndebs** [comment]
Hello Michael,

Thank you for your feedback! 

I correctly launch the code that you mentioned, so the mistake was somewhere else. I finally found my problem: in file load.py, l.85 https://github.com/MIC-DKFZ/nnDetection/blob/61d46a7fac4ed754f8c3c27709ad50dbe224ac33/nndet/io/load.py#L85 :

``` 
seg_props_file = f"{str(seg_file).split('.')[0]}.json"

```

It turns out that the path of my json file contained dots ('/datadrive/db_1.8/Task000D3_Example/).  Therefore the split only retained the first part of my path ('/datadrive/db_1.'). 

I replaced this line by the following 3 lines and it works:

``` 
head,tail = os.path.split(seg_file)
seg_props_file_name = f"{str(tail).split('.')[0]}.json"
seg_props_file = os.path.join(head,seg_props_file_name)
``` 
Best,

Noëlie
